src/binary-reader-ir.cc

```diff
@@ -687,11 +729,62 @@ Result BinaryReaderIR::OnStoreExpr(Opcode opcode,
   return AppendExpr(expr);
 }
 
+Result BinaryReaderIR::OnThrowExpr(Index except_index) {
+  return AppendExpr(new ThrowExpr(Var(except_index)));
+}
+
 Result BinaryReaderIR::OnTeeLocalExpr(Index local_index) {
   auto expr = new TeeLocalExpr(Var(local_index, GetLocation()));
   return AppendExpr(expr);
 }
 
+Result BinaryReaderIR::OnTryExpr(Index num_types, Type* sig_types) {
+  auto expr = new TryExpr();
+  expr->block = new Block();
+  expr->block->sig.assign(sig_types, sig_types + num_types);
+  if (Failed(AppendExpr(expr))) {
+    delete expr;
+    return Result::Error;
+  }
+  PushLabel(LabelType::Try, &expr->block->exprs);
+  LabelNode* label;
+  if (Failed(TopLabel(&label))) {
+    PrintError("Internal error installing try block");
```

Comment:
I agree, if PushLabel() is written correctly, this should succeed. Changing to assert.